autoload/ale/c.vim

```diff
@@ -91,3 +92,27 @@ function! ale#c#FindCompileCommands(buffer) abort
 
     return ''
 endfunction
+
+" Given a buffer number, return the relevant command from
+" compile_commands.json. Return 0 if none could be found.
+function! ale#c#GetCompileCommand(buffer) abort
+    let l:build_dir = ale#c#FindCompileCommands(a:buffer)
+    if empty(l:build_dir)
+        return 0
+    endif
+    let l:commands = json_decode(readfile(l:build_dir . '/compile_commands.json'))
+    let l:file = fnamemodify(bufname(a:buffer), ':p')
+    " save current directory, so we can restore it before returning
+    let l:cwd = getcwd()
+    for l:command in l:commands
+        " temporarily change dir to run fnamemodify
+        execute 'cd' l:command.directory
+        let l:command.file = fnamemodify(l:command.file, ':p')
+        if l:file ==# l:command.file
```

Comment:
For strings, I recommend using `is#` instead, as `'x' ==# 0`, which can be source of bugs.